packages/react-examples/src/react-components/FluentUIComponents/StylingComponents.stories.mdx

```diff
@@ -0,0 +1,120 @@
+import { Meta, Source } from '@storybook/addon-docs/blocks';
+
+<Meta title="Concepts/Styling Components" />
+
+## Styling components
+
+To style Fluent UI components `makeStyles` is used.
+
+`makeStyles` is a proprietary CSS-in-JS implementation which generates atomic CSS classes and can be imported from "@fluentui/react-components".
+
+To style a component, you need to call `makeStyles` in a module scope to create a React hook to be used inside a component.
+
+The `makeStyles` call accepts an object of items where each key is a unique identifier and each value is an object with styles.
+
+<!-- TODO: format as <Code /> -->
+
+```jsx
+import { makeStyles } from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: { color: 'red' },
+});
+
+function Component() {
+  const classes = useStyles();
+
+  return <div className={classes.root} />;
+}
+```
+
+### Merging component styles
+
+There are cases where you need to merge classes from multiple `useStyles` calls. **It is not possible to just concatenate the classes!**
+
+To properly merge the classes, you need to use `ax()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.
+
+When `ax()` is called, it merges all classes from top to bottom - the latter argument overwrites the previous ones (similar to `Object.assign()`).
+
+```jsx
+const useStyles = makeStyles({
+  blueBold: {
+    color: 'blue',
+    fontWeight: 'bold',
+  },
+  red: {
+    color: 'red',
+  },
+});
+
+function Component() {
+  const classes = useStyles();
+
+  const first = ax(classes.blueBold, classes.red); // { color: 'red', fontWeight: 'bold' }
+  const second = ax(classes.red, classes.blueBold); // { color: 'blue', fontWeight: 'bold' }
+}
+```
+
+### Mapping props and state to styles
+
+Both `makeStyles` and `ax` are props and state agnostic. It is developers responsibility to, based on props and state, merge and apply appropriate parts of the `useStyles` call:
+
+```jsx
+const useStyles = makeStyles({
+  root: { color: 'red' },
+  rootPrimary: { color: 'blue' },
+});
+
+function Component(props) {
+  const classes = useStyles();
+
+  return <div className={ax('ui-component', classes.root, props.primary && classes.rootPrimary)} />;
+}
+```
+
+### Applying classes passed from parent
+
+The same approach is used to apply classes passed from parent component. Again, you need to use `ax()` to properly merge and deduplicate the classes.
```

Comment:
```suggestion
The same approach is used to apply classes passed from parent component. Again, you need to use `ax()` to properly merge and deduplicate the classes. Note again, that the order of application is important in `ax()` so that parent classes overwrite component classes.
```